### PR TITLE
New list game seats

### PR DESCRIPTION
**Pull request by EGimenoS**
Just a little fix to open_games. I assume that we want to show games that has been created but still doesn't have a black_player_id. If I understood wrong what open_games means please just discard the pull request.

### CODE REVIEW

**Comment by mbrown78** [comment]
Not quite it would also need to be games that the white player is also not apart of.  I suggest we put the open games feature to the side for moment and merge the changes for new_list_game seats. I will push the version with open games removed up and make a pull request.

**Comment by saakuffo** [comment]
started working on incorporating the master into this branch. Have some comflicts and build errors to go through doing this.

**Comment by saakuffo** [comment]
We've been paring on this for a bit. Code is a bit hacky right now, concerns are not cleanly separated, noted for refactoring

**Comment by saakuffo** [comment]
Got it to pass checks so I could remove that weird error. Writing the tests for this now.

File: app/controllers/dashboards_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class DashboardsController < ApplicationController
+
+    def show
+      @games = Game.all
+      @current_player_games = current_player_games
+      # @opponent_games = opponent_games
```
Comment (lasley) [inline app/controllers/dashboards_controller.rb]: Remove useless code

File: app/controllers/dashboards_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    end
+
+    def current_player_games
+      current_player = find_current_player
+      @current_player_games = current_player.games.limit(10)
+      # Game.where("(white_player_id = #{current_player.id} OR black_player_id = #{current_player.id})
```
Comment (lasley) [inline app/controllers/dashboards_controller.rb]: Remove useless code

File: app/controllers/dashboards_controller.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class DashboardsController < ApplicationController
+
+    def show
+      @games = Game.all
+      @current_player_games = current_player_games
```
Comment (lasley) [inline app/controllers/dashboards_controller.rb]: Does this work? Where does `current_player_games` get instantiated?

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -2,4 +2,8 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  def find_current_player
+    return Player.find(current_user.id)
```
Comment (lasley) [inline app/controllers/application_controller.rb]: Won't this give multiple players? Limit to the current game maybe?

File: app/controllers/games_controller.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
+class GamesController < ApplicationController
+
+    def index
+    end
+
+    def new
+      current_player = find_current_player
```
Comment (lasley) [inline app/controllers/games_controller.rb]: Will you have a player before the game exists? Maybe create the first player with the game, then second when the other user joins?

File: app/controllers/games_controller.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      @game = Game.new
+    end
+
+    def create
```
Comment (lasley) [inline app/controllers/games_controller.rb]: New and create? IMO we're duplicating the C part of CRUD. Way to combine the methods?

File: app/controllers/players_controller.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
+class PlayersController < ApplicationController
+
+
```
Comment (lasley) [inline app/controllers/players_controller.rb]: Nitpick - remove this extra whitespace/carriage return

File: app/controllers/players_controller.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  def new
+    @player = Player.new
+  end
+
+  def create
```
Comment (lasley) [inline app/controllers/players_controller.rb]: Why new and create?

File: app/models/game.rb [hunk 9]
```diff
@@ -1,4 +1,7 @@
 class Game < ActiveRecord::Base
-  belongs_to :player
-  has_many :pieces
+
+    belongs_to :player
+    has_many :pieces
+
+
```
Comment (lasley) [inline app/models/game.rb]: Nitpick - remove extra whitespace. Should only need one carriage return to separate things inside of classes/modules/methods. Only top level items need to carriage return separation (such as multiple classes in one file)

File: app/views/dashboards/show.html.erb [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+  <div>
+    <h1><%=game.game_title%></h1>
+    <%= link_to "Play this game", game_path(game) %>
+  </div>
+<% end %>
+
+
+<h1>Join a Game</h1>
+
+<% @games.each do |game| %>
```
Comment (lasley) [inline app/views/dashboards/show.html.erb]: All games, or just the ones with open seats?

File: app/views/games/new.html.erb [hunk 11]
```diff
@@ -0,0 +1,8 @@
+<h1> a simple form create new game </h1>
```
Comment (lasley) [inline app/views/games/new.html.erb]: Is this the title we want?

File: app/views/games/new.html.erb [hunk 12]
```diff
@@ -0,0 +1,8 @@
...
+
+<h1>Create your game</h1>
+<%= simple_form_for @game do |f| %>
+  <%= f.input :game_title %>
+  <%= f.input :white_player_id, :value => current_user.id, :as => :hidden %>
+  <%= f.submit 'Create', class: 'btn btn-danger' %>
+<% end %>
```
Comment (lasley) [inline app/views/games/new.html.erb]: http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: app/views/layouts/application.html.erb [hunk 13]
```diff
@@ -8,6 +8,39 @@
...
+  <div class="container-fluid">
+    <!-- Brand and toggle get grouped for better mobile display -->
+    <div class="navbar-header">
+      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
+        <span class="sr-only">Toggle navigation</span>
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>
+      </button>
+      <%= link_to 'Brand', root_path, class: 'navbar-brand' %>
```
Comment (lasley) [inline app/views/layouts/application.html.erb]: Might want to switch `Brand` to something meaningful, like `Home` or something.

File: app/views/layouts/application.html.erb [hunk 14]
```diff
@@ -18,5 +51,3 @@
 
 <%= yield %>
 
-</body>
-</html>
```
Comment (lasley) [inline app/views/layouts/application.html.erb]: Why remove these tags? They are required per HTML 4 & 5 RFC

File: app/views/players/new.html.erb [hunk 15]
```diff
@@ -0,0 +1,5 @@
+<h1>Choose your name</h1>
+<%= simple_form_for @player do |f| %>
+  <%= f.input :name %>
+  <%= f.submit 'Create', class: 'btn btn-danger' %>
+<% end %>
```
Comment (lasley) [inline app/views/players/new.html.erb]: http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

**Comment by saakuffo** [comment]
Taking over this branch and cleaning it up and writing tests.

File: app/assets/stylesheets/master.css.scss [hunk 16]
```diff
@@ -0,0 +1,24 @@
+@import "bootstrap-sprockets";
+@import "bootstrap";
+
+
+.editp {
```
Comment (saakuffo) [inline app/assets/stylesheets/master.css.scss]: Looks like this was meant for another feature. removed from this branch

File: app/controllers/dashboards_controller.rb (hunk 3 above)
Comment (saakuffo) [inline app/controllers/dashboards_controller.rb]: It does work but I changed how the code work. I now just pass all games to the page and then use some erb logic to decide what to show.

File: app/controllers/dashboards_controller.rb (hunk 1 above)
Comment (saakuffo) [inline app/controllers/dashboards_controller.rb]: done

File: app/controllers/dashboards_controller.rb (hunk 2 above)
Comment (saakuffo) [inline app/controllers/dashboards_controller.rb]: done

File: app/controllers/application_controller.rb (hunk 4 above)
Comment (saakuffo) [inline app/controllers/application_controller.rb]: Player's user_id column now has a validation for uniqueness. There should only ever be one user_id per player.

File: app/controllers/games_controller.rb (hunk 5 above)
Comment (saakuffo) [inline app/controllers/games_controller.rb]: Games can no longer be created if player does not exist for the current user.

File: app/assets/stylesheets/dashboards.scss [hunk 17]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the dashboards controller here.
```
Comment (lasley) [inline app/assets/stylesheets/dashboards.scss]: dashboard.scss and dashboards.scss? Seems duplicate

File: app/controllers/players_controller.rb (hunk 8 above)
Comment (lasley) [inline app/controllers/players_controller.rb]: This question remains unanswered/resolved

File: app/views/dashboards/show.html.erb [hunk 18]
```diff
@@ -0,0 +1,15 @@
+<h1>My Dashboard</h1>
+
+<h1>My Games</h1>
```
Comment (lasley) [inline app/views/dashboards/show.html.erb]: I would step down to an h2 here

File: app/views/games/new.html.erb [hunk 19]
```diff
@@ -0,0 +1,7 @@
+<h1>Welcome <%= current_user.player.name %>!</h1>
+<h1>Create your game</h1>
```
Comment (lasley) [inline app/views/games/new.html.erb]: I would step down to an h2 here

File: app/views/games/new.html.erb (hunk 12 above)
Comment (lasley) [inline app/views/games/new.html.erb]: This remains unresolved

File: app/views/layouts/application.html.erb (hunk 14 above)
Comment (lasley) [inline app/views/layouts/application.html.erb]: This remains unanswered/unresolved

File: app/views/players/new.html.erb (hunk 15 above)
Comment (lasley) [inline app/views/players/new.html.erb]: unresolved

File: app/controllers/games_controller.rb (hunk 6 above)
Comment (saakuffo) [inline app/controllers/games_controller.rb]: Dave maybe I am missing something here. This is how the course teaches use to create things. The new is a page we go to and send the empty new Game variable to be loaded with data and the create is what sends it to the Database to be committed.

Here is an example of the pattern from the course:
  def new
    @course = Course.new
  end

  def create
    @course = current_user.courses.create(course_params)
    redirect_to instructor_course_path(@course)
  end

File: app/controllers/players_controller.rb (hunk 8 above)
Comment (saakuffo) [inline app/controllers/players_controller.rb]: See the above. This is a pattern the course teaches us. If the pattern is wrong, please let me know why.

I get the feeling the course is pushing us towards treat each route as it's own specific thing. Additional separations of concerns and what not.

File: app/views/dashboards/show.html.erb (hunk 10 above)
Comment (saakuffo) [inline app/views/dashboards/show.html.erb]: All games. I decided to do the logic to parse for what games I am looking for on the page.
This has the internal assumption of how many users how app might actually have.

If you think it's a big enough of an issue, I will filter.

File: app/views/games/new.html.erb (hunk 11 above)
Comment (saakuffo) [inline app/views/games/new.html.erb]: Will change. most of the page displays are token descriptors. You can think of ALL the games pages as MV but not yet product level.

File: app/views/games/new.html.erb (hunk 12 above)
Comment (saakuffo) [inline app/views/games/new.html.erb]: will do. assumed This was one of the things rubocop fixed.

File: app/views/layouts/application.html.erb (hunk 14 above)
Comment (saakuffo) [inline app/views/layouts/application.html.erb]: didn't see this. will change it.

File: app/views/dashboards/show.html.erb (hunk 18 above)
Comment (saakuffo) [inline app/views/dashboards/show.html.erb]: Done

File: app/views/games/new.html.erb (hunk 19 above)
Comment (saakuffo) [inline app/views/games/new.html.erb]: done. pending upload.

File: app/views/players/new.html.erb (hunk 15 above)
Comment (saakuffo) [inline app/views/players/new.html.erb]: Done

File: app/views/games/new.html.erb (hunk 12 above)
Comment (saakuffo) [inline app/views/games/new.html.erb]: Done

File: app/controllers/players_controller.rb (hunk 8 above)
Comment (saakuffo) [inline app/controllers/players_controller.rb]: Leaving as is based on conversation in Slack. I.E. This is most likely a rails pattern the course teaches us.

**Pull request merged**